python/pylibcudf/pylibcudf/io/orc.pyx

```diff
@@ -0,0 +1,283 @@
+                    ts_stats.maximum_utc.value() / 1000, datetime.timezone.utc
+                )
+        else:
+            raise ValueError("Unsupported statistics type")
+
+    def __getitem__(self, item):
+        return self.column_stats[item]
+
+    def __contains__(self, item):
+        return item in self.column_stats
+
+    def get(self, item, alt=None):
+        return self.column_stats.get(item, alt)
+
+    @staticmethod
+    cdef OrcColumnStatistics from_libcudf(column_statistics& col_stats):
+        cdef OrcColumnStatistics out = OrcColumnStatistics.__new__(OrcColumnStatistics)
+        out.number_of_values_c = col_stats.number_of_values
+        out.has_null_c = col_stats.has_null
+        out.type_specific_stats_c = col_stats.type_specific_stats
+        out._init_stats_dict()
+        return out
+
+
+cdef class ParsedOrcStatistics:
+
+    @property
+    def column_names(self):
+        return [name.decode() for name in self.c_obj.column_names]
+
+    @property
+    def file_stats(self):
+        stats_lst = []
+        for i in range(self.c_obj.file_stats.size()):
+            stats_lst.append(OrcColumnStatistics.from_libcudf(self.c_obj.file_stats[i]))
+        return stats_lst
+
+    @property
+    def stripes_stats(self):
+        stats_lst = []
+        for stripe_stats_c in self.c_obj.stripes_stats:
+            stripe_stats = []
+            for i in range(stripe_stats_c.size()):
+                stripe_stats.append(OrcColumnStatistics.from_libcudf(stripe_stats_c[i]))
+            stats_lst.append(stripe_stats)
+        return stats_lst
+
+    @staticmethod
+    cdef ParsedOrcStatistics from_libcudf(parsed_orc_statistics& orc_stats):
+        cdef ParsedOrcStatistics out = ParsedOrcStatistics.__new__(ParsedOrcStatistics)
+        out.c_obj = move(orc_stats)
+        return out
+
+
+cpdef TableWithMetadata read_orc(
+    SourceInfo source_info,
+    list columns = None,
+    list stripes = None,
+    size_type skip_rows = 0,
+    size_type nrows = -1,
+    bool use_index = True,
+    bool use_np_dtypes = True,
+    DataType timestamp_type = DataType(type_id.EMPTY),
+    list decimal128_columns = None,
+):
+    """Reads an ORC file into a :py:class:`~.types.TableWithMetadata`.
+
+    Parameters
+    ----------
+    source_info : SourceInfo
+        The SourceInfo object to read the Parquet file from.
+    columns : list, default None
+        The string names of the columns to be read.
+    stripes : list[list[size_type]], default None
+        List of stripes to be read.
+    skip_rows : int64_t, default 0
+        The number of rows to skip from the start of the file.
+    nrows : size_type, default -1
+        The number of rows to read. By default, read the entire file.
+    use_index : bool, default True
+        Whether to use the row index to speed up reading.
+    use_np_dtypes : bool, default True
+        Whether to use numpy compatible dtypes.
```

Comment:
Suggesting some missing parameter docs -- this is a total guess for what these do, so please revise accordingly.
```suggestion
    use_np_dtypes : bool, default True
        Whether to use numpy compatible dtypes.
    timestamp_type : DataType, default EMPTY
        Type to use for timestamps. (???)
    decimal128_columns : list[size_type]
        Indices of decimal128 columns. (???)
```